iphone/cli/hooks/run.js

```diff
@@ -30,102 +22,176 @@ exports.init = function (logger, config, cli) {
+			const trimRE = new RegExp('^.*' + builder.tiapp.name + '\\[[^\\]]+\\]\\s*', 'g');
+			const logLevelRE = new RegExp('^(\u001b\\[\\d+m)?\\[?(' + levels.join('|') + '|log|timestamp)\\]?\\s*(\u001b\\[\\d+m)?(.*)', 'i');
+			function handleLogFile(line) {
+				// Titanium app log messages
+				if (!simStarted) {
+					simStarted = true;
+					logger.log(('-- ' + startLogTxt + ' ' + (new Array(75 - startLogTxt.length)).join('-')).grey);
+				}
+				line = line.replace(trimRE, '');
+				const m = line.match(logLevelRE);
+				if (m) {
+					lastLogger = m[2].toLowerCase();
+					line = m[4].trim();
+				}
+				if (levels.indexOf(lastLogger) === -1) {
+					logger.log(('[' + lastLogger.toUpperCase() + '] ').cyan + line);
+				} else {
+					logger[lastLogger](line);
+				}
+			}
+
+			if (cli.argv.target === 'macos') {
+				startLogTxt = __('Start mac application log');
+				endLogTxt = __('End mac application log');
+
+				const HOME = require('os').homedir();
+				const logFile = path.join(HOME, 'Documents', `${builder.tiapp.guid}.log`);
+
+				// TODO: It does force quit. Find another way to quit.
+				exec(`pkill -QUIT -x ${builder.tiapp.name}`, function () {
+					// If there's already a log file from previous run, wipe it
+					if (fs.pathExistsSync(logFile)) {
+						try {
+							fs.accessSync(logFile, fs.constants.W_OK);
+							fs.removeSync(logFile);
+						} catch (err) {
+							// Note to user that they process does not have rights to the log file, which typically means macos
+							// Catalina has prompted and been denied access to the user's Documents folder!
+							// they'll need to add node (or whatever host process) rights to the Documents folder
+							logger.error('This process does not have rights to delete the existing log file. Please ensure you have not denied rights to your Documents folder, as it may result in not being able to see the app logs. If so, you will need to allow access in the System Preferences > Security & Privacy > Files and Folders UI.');
+						}
 					}
-				})
-				.on('log', function (msg, simHandle) {
-					// system log messages
-					logger.trace(('[' + simHandle.appName + '] ' + msg).grey);
-				})
-				.on('log-debug', function (msg) {
-					// ioslib debug messages
-					logger.trace(('[ioslib] ' + msg.replace('[DEBUG] ', '')).grey);
-				})
-				.on('log-error', function (msg, simHandle) {
-					// system log error messages
-					logger.error('[' + simHandle.appName + '] ' + msg);
-				})
-				.on('app-started', function () {
+
+					// Open the app
+					logger.info(__('Launching Mac application'));
+					const child = spawn(`${builder.iosBuildDir}/${builder.tiapp.name}.app/Contents/MacOS/${builder.tiapp.name}`);
+					child.on('error', err => logger.error(err));
+					// "Forward" the exit code of the app to this process (when the app exits)
+					child.on('exit', (code, signal) => {
+						endLog();
+						process.exit(code);
+					});
+					child.on('close', (code, signal) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  iphone/cli/hooks/...
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/hooks/run.js;
  Line: 112;
-->

- :warning: iphone/cli/hooks/run.js line 112 – 'signal' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)